test/unit_tests/test_helper_functions.jl

```diff
@@ -13,3 +13,6 @@ NProgMoms = [2, 2, 3]
 @test get_dist_moments_ind_range(NProgMoms, 1) == 1:2
 @test get_dist_moments_ind_range(NProgMoms, 3) == 5:7
 @test_throws Exception get_dist_moments_ind_range(NProgMoms, 4)
+
+norm_factors = get_moments_normalizing_factors(NProgMoms, [10.0, 0.1])
+@test norm_factors ≈ [10.0, 1.0, 10.0, 1.0, 10.0, 1.0, 0.1] atol = 1e-12
```

Comment:
Some other outstanding tests we are missing, if you want to add them in this PR:

1. Opt tests on analytical coalescence functions
2. Opt tests on KernelTensors

Reply:
I added the opt tests